src/config.c

```diff
@@ -931,10 +938,14 @@ void configSetCommand(client *c) {
     }
 end:
     zfree(config_names);
-    dictRelease(old_values);
-    dictRelease(set_configs);
-    dictRelease(new_values);
     dictRelease(apply_fns);
+    di = dictGetIterator(set_config_values);
+    while ((de = dictNext(di)) != NULL) {
+        configSetValue *config_value = dictGetVal(de);
+        sdsfree(config_value->old_value);
+    }
```

Comment:
Can't we just create the appropriate `valDestructor` for our dict and avoid this loop?

Reply:
Yes, just added that to the code now.